integrations/weaviate/src/haystack_integrations/document_stores/weaviate/_filters.py

```diff
@@ -0,0 +1,290 @@
+
+    return inverted_condition
+
+
+def _parse_logical_condition(condition: Dict[str, Any]) -> Dict[str, Any]:
+    if "operator" not in condition:
+        msg = f"'operator' key missing in {condition}"
+        raise FilterError(msg)
+    if "conditions" not in condition:
+        msg = f"'conditions' key missing in {condition}"
+        raise FilterError(msg)
+
+    operator = condition["operator"]
+    if operator in ["AND", "OR"]:
+        return {
+            "operator": operator.lower().capitalize(),
+            "operands": [_parse_comparison_condition(c) for c in condition["conditions"]],
+        }
+    elif operator == "NOT":
+        inverted_conditions = _invert_condition(condition)
+        return _parse_logical_condition(inverted_conditions)
+    else:
+        msg = f"Unknown logical operator '{operator}'"
+        raise FilterError(msg)
+
+
+def _infer_value_type(value: Any) -> str:
+    if value is None:
+        return "valueNull"
+
+    if isinstance(value, bool):
+        return "valueBoolean"
+    if isinstance(value, int):
+        return "valueInt"
+    if isinstance(value, float):
+        return "valueNumber"
+
+    if isinstance(value, str):
+        try:
+            parser.isoparse(value)
+            return "valueDate"
+        except ValueError:
+            return "valueText"
+
+    msg = f"Unknown value type {type(value)}"
+    raise FilterError(msg)
+
+
+def _handle_date(value: Any) -> str:
+    if isinstance(value, str):
+        try:
+            return parser.isoparse(value).strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        except ValueError:
+            pass
+    return value
+
+
+def _equal(field: str, value: Any) -> Dict[str, Any]:
+    if value is None:
+        return {"path": field, "operator": "IsNull", "valueBoolean": True}
+    return {"path": field, "operator": "Equal", _infer_value_type(value): _handle_date(value)}
+
+
+def _not_equal(field: str, value: Any) -> Dict[str, Any]:
+    if value is None:
+        return {"path": field, "operator": "IsNull", "valueBoolean": False}
+    return {
+        "operator": "Or",
+        "operands": [
+            {"path": field, "operator": "NotEqual", _infer_value_type(value): _handle_date(value)},
+            {"path": field, "operator": "IsNull", "valueBoolean": True},
+        ],
+    }
+
+
+def _greater_than(field: str, value: Any) -> Dict[str, Any]:
+    if value is None:
+        # When the value is None and '>' is used we create a filter that would return a Document
+        # if it has a field set and not set at the same time.
+        # This will cause the filter to match no Document.
+        # This way we keep the behavior consistent with other Document Stores.
+        return {
+            "operator": "And",
+            "operands": [
+                {"path": field, "operator": "IsNull", "valueBoolean": False},
+                {"path": field, "operator": "IsNull", "valueBoolean": True},
+            ],
+        }
```

Comment:
If I understood correctly, this is a trick to create a filter that returns no Documents.
It is done to keep the global filtering logic unaffected. Right?

If so, can we centralize that logic in a utility function?